go/arrow/array/concat_test.go

```diff
@@ -158,6 +159,32 @@ func (cts *ConcatTestSuite) generateArr(size int64, nullprob float64) arrow.Arra
 		bldr := array.NewListBuilder(memory.DefaultAllocator, arrow.PrimitiveTypes.Int8)
 		defer bldr.Release()
 
+		valid := make([]bool, len(offsetsVector)-1)
+		for i := range valid {
+			valid[i] = true
+		}
+		bldr.AppendValues(offsetsVector, valid)
+		vb := bldr.ValueBuilder().(*array.Int8Builder)
+		for i := 0; i < values.Len(); i++ {
+			if values.IsValid(i) {
+				vb.Append(values.Value(i))
+			} else {
+				vb.AppendNull()
+			}
+		}
+		return bldr.NewArray()
+	case arrow.LARGE_LIST:
+		valuesSize := size * 8
```

Comment:
yea, I'm doing * 8 here since it's expected to be 64-bit integers for the test

Reply:
Well, I understand that `cts.largeOffsets` generated `size` 64-bit offsets between 0 and `valuesSize` (inclusive).
But why does the number of list offsets be equal to 8 times the number of values in the child array?

Reply:
`size` is the number of elements in the list array (which is why `cts.largeOffsets` outputs a slice with `size+1` elements because offsets should be 1 + #of elems). I multiplied by 8 here just to have a larger child array than I was using with the initial List cases, thus the lengths of the list elements will be *on average* larger than the previous case.

TL;DR: `size` is the number of list elements (and `size+1` is the number of offsets). `valuesSize` is the total length of the child array which will get sliced up into those list elements randomly.